app/assets/javascripts/map/components/registration-form.js

```diff
@@ -61,13 +70,15 @@ function RegistrationForm() {
             onchange: event => { data.questions[question.slug] = event.currentTarget.value },
           })
         }),
-        alert ? m('.sya-registration__message', { class: alert.status }, alert.message) : null,
+        alert ? m('.sya-registration__message.error', { class: alert.status }, alert.message) : null,
         m('.sya-registration__notice',
           m.trust(Util.translate('registration.notice.text', {
             link: `<a class="sya-registration__notice-link" href="/${AtlasApp.config.locale}/privacy" target="_blank">${Util.translate('registration.notice.link')}</a>`
           }))
         ),
-        m('button.sya-registration__submit', Util.translate('registration.form.submit'))
+        m('button.sya-registration__submit',{
+          disabled: alert?.message === Util.translate('registration.form.email_invalid') || !data?.name || !data.email
```

Comment:
We should use the status variable to check what type of response got.
```javascript
alert?.status == 'error' || !data?.name || !data.email
```

Did you test that after an error appears it is possible to clear the error by changing an input?

Reply:
if i apply the above condition the code will become disable in case of any other error